pages/challengeDetail/[id].jsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import Grid from "@mui/material/Grid";
+import { useChallengeContext } from '../../ChallengeContext'
+import Challenge from "@/components/Challenge";
+import GoBackLink from "@/components/GoBackLink";
+
+const challengeDetail = () => {
+  const { data } = useChallengeContext()
+
+  return (
+    <Grid
+      container
+      spacing={4}
+      direction="column"
+      sx={{ marginTop: 2, }}
+    >
+      <Grid item>
+        <GoBackLink />
+      </Grid>
+      <Grid item xs={12} md={12}>
+        {data && (<Challenge challenge={data} />)}
+      </Grid>
+    </Grid>
+  )
+}
+
+export default challengeDetail
```

Comment:
> We can also use the conventional shorthand syntax here:

Could you tell me why you prefer the regular function over the arrow function? If we are going to add TS soon, we should keep using the arrow function so we can strictly use type React.FC

Reply:
Sure, I can clarify this.

I tend to use this pattern since it's convention and recommended by the React core team (see previously mentioned docs).

Even when using TypeScript, [It can be worthwhile to prefer NOT using return types](https://youtu.be/I6V2FkW1ozQ) and [`React.FC` can mostly be avoided anyways](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components/) especially provided the context of inferred types.

Reply:
> Sure, I can clarify this.
> 
> I tend to use this pattern since it's convention and recommended by the React core team (see previously mentioned docs).
> 
> Even when using TypeScript, [It can be worthwhile to prefer NOT using return types](https://youtu.be/I6V2FkW1ozQ) and [`React.FC` can mostly be avoided anyways](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components/) especially provided the context of inferred types.

TY for the docs, and I learned that we don't need to use `React.FC` in Next.js. Yet, IMO I still prefer to use the arrow function because of its performance in Next.js (especially for pages) and readability.

Reply:
Oh, I didn't know arrow functions had improved performance in Next.js. Can you point to something that backs this up? I'm genuinely interested to learn more about this.

In terms of readability, we can talk more about our opinions around that so we can keep the code consistent

Reply:
Ok, apologies for the confusion, I misunderstood. This isn't related to performance issues but to preferences. I'm okay with changing them back to rewrite all functional components to follow the convention.